src/components/Pages/ArtistPage/ArtistPage.test.js

```diff
@@ -0,0 +1,165 @@
+  })
+  it('renders correctly', () => {
+    const { container } = render(
+      <ArtistPage
+        artists={[
+          {
+            artist: 'Unprocessed',
+            songTitle: 'Real',
+            songSnippet: 'https://test/',
+            id: 'a',
+          },
+        ]}
+        toggleCurrentSongId={() => {}}
+        isSongPlaying={false}
+        currentSongId="a"
+        saveSong={() => {}}
+        savedSongs={[]}
+      />
+    )
+    expect(container.firstChild).toMatchSnapshot()
+  })
+  it('sets the current id to currentSongId', () => {
+    const toggleCurrentSongIdMock = jest.fn()
+    const { getByTestId } = render(
+      <ArtistPage
+        artists={[
+          {
+            artist: 'Unprocessed',
+            songTitle: 'Real',
+            songSnippet: 'https://test/',
+            id: 'a',
+          },
+        ]}
+        toggleCurrentSongId={toggleCurrentSongIdMock}
+        isSongPlaying={false}
+        currentSongId="a"
+        saveSong={() => {}}
+        savedSongs={[]}
+      />
+    )
+    const button = getByTestId('audio-button')
+    user.click(button)
+    expect(toggleCurrentSongIdMock).toHaveBeenCalled()
+  })
+  it('toggles the isSongPlaying state correctly', () => {
+    const { getByTitle, queryByTitle, rerender } = render(
+      <ArtistPage
+        artists={[
+          {
+            artist: 'Unprocessed',
+            songTitle: 'Real',
+            songSnippet: 'https://test/',
+            id: 'a',
+          },
+        ]}
+        toggleCurrentSongId={() => {}}
+        isSongPlaying={false}
+        currentSongId="a"
+        saveSong={() => {}}
+        savedSongs={[]}
+      />
+    )
+    expect(getByTitle('play')).toBeInTheDocument()
+
+    rerender(
+      <ArtistPage
+        artists={[
+          {
+            artist: 'Unprocessed',
+            songTitle: 'Real',
+            songSnippet: 'https://test/',
+            id: 'a',
+          },
+        ]}
+        toggleCurrentSongId={() => {}}
+        isSongPlaying={true}
+        currentSongId="a"
+        saveSong={() => {}}
+        savedSongs={[]}
+      />
+    )
+    expect(queryByTitle('play')).not.toBeInTheDocument()
+    expect(getByTitle('pause')).toBeInTheDocument()
+  })
+  it('calls the saveSong function correctly', () => {
+    const saveSongMock = jest.fn()
+    const { getByTestId } = render(
+      <ArtistPage
+        artists={[
+          {
+            artist: 'Unprocessed',
+            songTitle: 'Real',
+            songSnippet: 'https://test/',
+            id: 'a',
+          },
+        ]}
+        toggleCurrentSongId={() => {}}
+        isSongPlaying={false}
+        currentSongId="a"
+        saveSong={saveSongMock}
+        savedSongs={[]}
+      />
+    )
+    const button = getByTestId('button')
+    user.click(button)
+    expect(saveSongMock).toHaveBeenCalled()
+  })
+  it('renders a new array for savedSongs by calling saveSong function', () => {
```

Comment:
I am little bit confused by this test, maybe I just do not get it though. :D 
You want to test, if a new array is rendered but you fill it yourself?